### ISSUE TITLE
Class level __getattr__ causes error: __getattr__ is not valid at the module level outside a stub file

### ISSUE THREAD

**Issue by stoklund** [python/mypy#4069]
After updating to mypy 0.530, I am getting an error on a `__getattr__()` method defined on a class:

```
$ mypy --version
mypy 0.530
$ python3 --version
Python 3.6.3
[~/cretonne/lib/cretonne/meta]
$ cat mypy.ini 
[mypy]
disallow_untyped_defs = True
warn_unused_ignores = True
warn_return_any = True
[~/cretonne/lib/cretonne/meta]
$ mypy cdsl/formats.py
cdsl/formats.py:143: error: __getattr__ is not valid at the module level outside a stub file
```

I don't have any module level `__getattr__()` functions.

This didn't reproduce with a smaller test case, but you can check out the [Cretonne repo](https://github.com/stoklund/cretonne) to reproduce.

**Comment by ethanhs** [comment]
Confirmed on master, I am stepping through now. I appears the issue only occurs in the second pass of the checker (the class is added to the scope stack in the first pass, but not in the second).

**Comment by gvanrossum** [comment]
Do you have a smaller repro yet?

**Comment by ilevkivskyi** [comment]
@ethanhs Your diagnosis seems right. Here is a simple repro:
```python
class C:
    def __getattr__(self, attr: str) -> int:
        x: F
        return x.f

class F:
    def __init__(self, f: int) -> None:
        self.f = f
```

**Comment by ilevkivskyi** [comment]
A related bug:
```python
from typing import TypeVar

T = TypeVar('T', bound=C)
class C:
    def m(self: T) -> T:
        class Inner:
            x: F
            f = x.f
        return self

class F:
    def __init__(self, f: int) -> None:
        self.f = f
```
results in a funny error:
```
main:5: error: The erased type of self '__main__.C' is not a supertype of its class '__main__.Inner@5'
```

**Comment by ilevkivskyi** [comment]
OK, here is the PR #4073.

**Comment by stoklund** [comment]
Confirmed fixed in mypy-0.540. Thanks for the quick response!
